### DISCUSSION TITLE
Ability to read PDF documents

### DISCUSSION THREAD

**Discussion by lilydelalande** [aaif-goose/goose#481] [Ideas]
Sometimes I'd like Goose to read PDF documents -- ideally we can transform the text of the PDF to strings goose can read like any other file, and also I'd like if any images in the PDF were also viewable to Goose

**Comment by salman1993** [comment]
Yes, we should have a good extension to do this. I wonder if there are already good MCP servers out there that do this. maybe this one? https://github.com/xraywu/mcp-pdf-extraction-server

**Comment by blackgirlbytes** [comment]
Closing as we've gotten this to work with a bundled extension called the PDF reader. People can learn how to use it here: https://block.github.io/goose/docs/tutorials/pdf-mcp
